libraries/rush-lib/src/cli/actions/DeployAction.ts

```diff
@@ -101,6 +103,21 @@ export class DeployAction extends BaseRushAction {
     });
   }
 
+  private _getDependencyProjects(project: RushConfigurationProject): Set<RushConfigurationProject> {
+    const projects: Set<RushConfigurationProject> = new Set();
+    const queue: RushConfigurationProject[] = [project];
+
+    while (queue.length > 0) {
+      const _project: RushConfigurationProject = queue.shift()!;
```

Comment:
did we forget to fill 'projects' in this function?

Reply:
+1

Reply:
`shift()` is an `O(n)` operation making this loop `O(n^2)`.  If N=1,000 then N^2=1,000,000 which might be significant

Reply:
Yes, deleted by mistake...